### PR TITLE
Implemented 3 new describe functions in erlcloud_ec2

### PR DESCRIPTION
**Pull request by SergeyOvchinnik**
Problem:

erlcloud_ec2 did not have support for the following actions:

- DescribeAccountAttributes
- DescribeNatGateways
- DescribeVpcPeeringConnections

Solution:

Implemented the functions to support those actions:

- describe_account_attributes
- describe_nat_gateways
- describe_vpc_peering_connections

Also made slight change to make names of extract functions more consistent

Reviewers: @motobob

### CODE REVIEW

**Comment by SergeyOvchinnik** [comment]
test it

File: src/erlcloud_ec2.erl [hunk 1]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+    ].
+    
+extract_peering_connection_vpc_info(Node) ->
+    [
+        {cidr_block, get_text("cidrBlock", Node)},
+        {ipv6_cidr_block_set,
+            [extract_ipv6_cidr_block(Item) || Item <- xmerl_xpath:string("ipv6CidrBlockSet/item", Node)]},
+        {owner_id, get_text("ownerId", Node)},
+        {peering_options,
+            extract_vpc_peering_connection_options_description(lists:nth(1, xmerl_xpath:string("peeringOptions", Node)))},
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: s/_description// 
it's just options.

File: src/erlcloud_ec2.erl [hunk 2]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+    [
+        {allow_egress_from_local_classic_link_to_remote_vpc,
+            get_bool("allowEgressFromLocalClassicLinkToRemoteVpc", Node)},
+        {allow_egress_from_local_vpc_to_remote_classic_link,
+            get_bool("allowEgressFromLocalVpcToRemoteClassicLink", Node)},
+        {allow_dns_resolution_from_remote_vpc,
+            get_bool("allowDnsResolutionFromRemoteVpc", Node)}
+    ].
+    
+extract_vpc_peering_connection_state_reason(Node) ->
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: s/_state_reason/_status

File: src/erlcloud_ec2.erl [hunk 3]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+    when is_list(VpcPeeringConnectionId), is_list(Filter) orelse Filter =:= none ->
+    describe_vpc_peering_connections(VpcPeeringConnectionId, Filter, default_config());
+describe_vpc_peering_connections(VpcPeeringConnectionId, Config)
+    when is_list(VpcPeeringConnectionId), is_record(Config, aws_config) ->
+    describe_vpc_peering_connections(VpcPeeringConnectionId, [], Config).
+
+
+-spec describe_vpc_peering_connections(vpc_peering_connection_id(), filter_list(), aws_config()) ->
+            ok_error([proplist()]).
+describe_vpc_peering_connections(VpcPeeringConnectionId, Filter, Config)
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: s/VpcPeeringConnectionId/VpcPeeringConnectionIds.
this is a list.

File: src/erlcloud_ec2.erl [hunk 4]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+    [
+        {provisioned, get_text("provisioned", Node)},
+        {provision_time, get_text("provisionTime", Node)},
+        {requested, get_text("requested", Node)},
+        {request_time, get_text("requestTime", Node)},
+        {status, get_text("status", Node)}
+    ].
+    
+-spec describe_vpc_peering_connections() -> ok_error([proplist()]).
+describe_vpc_peering_connections() ->
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: please put links to docs for every api

File: src/erlcloud_ec2.erl [hunk 5]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+            {ok, AccountAttributes};
+        {error,  _} = E -> E
+    end.
+    
+extract_vpc_peering_connection(Node) ->
+    [
+        {accepter_vpc_info, extract_peering_connection_vpc_info(lists:nth(1, xmerl_xpath:string("accepterVpcInfo", Node)))},
+        {expiration_time, get_text("expirationTime", Node)},
+        {requester_vpc_info, extract_peering_connection_vpc_info(lists:nth(1, xmerl_xpath:string("requesterVpcInfo", Node)))},
+        {status, extract_vpc_peering_connection_state_reason(lists:nth(1, xmerl_xpath:string("status", Node)))},
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: status and many other are optional and 
```
lists:nth(1, []).
** exception error: no function clause matching lists:nth(1,[]) (lists.erl, line 170)
```
please rework

File: src/erlcloud_ec2.erl [hunk 6]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+extract_nat_gateway(Node) ->
+    [
+        {create_time, get_text("createTime", Node)},
+        {delete_time, get_text("deleteTime", Node)},
+        {failure_code, get_text("failureCode", Node)},
+        {failure_message, get_text("failureMessage", Node)},
+        {nat_gateway_address_set,
+            [extract_nat_gateway_address(Item) || Item <- xmerl_xpath:string("natGatewayAddressSet/item", Node)]},
+        {nat_gateway_id, get_text("natGatewayId", Node)},
+        {provisioned_bandwidth, extract_provisioned_bandwidth(lists:nth(1, xmerl_xpath:string("provisionedBandwidth", Node)))},
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: this is an optional field

File: src/erlcloud_ec2.erl [hunk 7]
```diff
@@ -3545,10 +3561,218 @@ extract_cgw(Node) ->
...
+         is_list(NextToken) orelse NextToken =:= undefined ->
+    describe_nat_gateways(NatGatewayId, Filter, MaxResults, NextToken, default_config());
+describe_nat_gateways(Filter, MaxResults, NextToken, Config)
+    when is_list(Filter) orelse Filter =:= none, is_integer(MaxResults),
+         is_list(NextToken) orelse NextToken =:= undefined, is_record(Config, aws_config) ->
+    describe_nat_gateways([], Filter, MaxResults, NextToken, Config).
+
+-spec describe_nat_gateways(nat_gateway_id(), filter_list(), ec2_max_result(), ec2_token(), aws_config())
+        -> ok_error([proplist()]).
+describe_nat_gateways(NatGatewayId, Filter, MaxResults, NextToken, Config)
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: s/NatGatewayId/NatGatewayIds
this is a list

File: src/erlcloud_ec2.erl [hunk 8]
```diff
@@ -3274,24 +3290,24 @@ describe_tags(Filters, MaxResults, NextToken, Config)
...
+-spec extract_tag_record(term()) -> #ec2_tag{}.
+extract_tag_record(Node) ->
     #ec2_tag{
              resource_id = get_text("resourceId", Node),
              resource_type = get_text("resourceType", Node),
              key = get_text("key", Node),
              value = get_text("value", Node)}.
 
-extract_tag_item(Node) ->
+extract_tag(Node) ->
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: please stop changing unnecessary stuff

**Comment by SergeyOvchinnik** [comment]
@motobob I've made the changes you proposed and removed some optional extracts, please take a look

File: src/erlcloud_ec2.erl [hunk 9]
```diff
@@ -233,6 +246,9 @@
 -type describe_spot_fleet_instances_return() ::
     {ok, [{instances, [proplist()]} | {next_token, string()}]} | {error, term()}.
 -type spot_fleet_instance_id() :: string().
+-type account_attribute_name() :: [string()].
+-type nat_gateway_id() :: [string()].
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: s/id/ids ;)

File: src/erlcloud_ec2.erl [hunk 10]
```diff
@@ -233,6 +246,9 @@
...
+-type vpc_peering_connection_id() :: [string()].
```
Comment (motobob) [inline src/erlcloud_ec2.erl]: same as above

**Comment by motobob** [comment]
@SergeyOvchinnik indeed comments were addressed but please fix specs and i'll merge after

**Pull request merged**